pandas/tests/frame/test_sorting.py

```diff
@@ -598,3 +598,67 @@ def test_sort_index_intervalindex(self):
             closed='right')
         result = result.columns.levels[1].categories
         tm.assert_index_equal(result, expected)
+
+    def test_sort_index_na_position_with_categories(self):
+        # GH 22556
+        # Positioning missing value properly when column is Categorical.
+        categories = ['A', 'B', 'C']
+        list_of_nans = [np.nan, np.nan]
+        df = pd.DataFrame({
+            'c': pd.Categorical(['A', np.nan, 'B', np.nan, 'C'],
+                                categories=categories,
+                                ordered=True)})
+        result_ascending_na_first = df.sort_values(by='c',
```

Comment:
rather than naming result / expected like this, can you just name them result & expected, and put a comment before the test case describing; its easier to read.

If you *can* easily parameterize things would be great as well (though may not be so easy).

Reply:
@jreback thanks for all the suggestions, I have made the changes except for the parameterization (couldn't figure out a good way to do that.. )